nilearn/plotting/html_surface.py

```diff
@@ -21,26 +21,46 @@ class SurfaceView(HTMLDocument):
 
 
 def _get_vertexcolor(surf_map, cmap, norm,
-                     absolute_threshold=None, bg_map=None):
+                     absolute_threshold=None, bg_map=None,
+                     bg_on_data=None, scale_bg_map=None, darkness=None):
     vertexcolor = cmap(norm(surf_map).data)
-    if absolute_threshold is None:
-        return to_color_strings(vertexcolor)
+
     if bg_map is None:
         bg_map = np.ones(len(surf_map)) * .5
         bg_vmin, bg_vmax = 0, 1
     else:
         bg_map = surface.load_surf_data(bg_map)
+
+    # scale background map if need be
+    if scale_bg_map:
         bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
-    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
-    bg_color = plt.get_cmap('Greys')(bg_norm(bg_map))
-    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
-        np.abs(surf_map) < absolute_threshold]
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_data = bg_norm(bg_map)
+    else:
+        bg_data = bg_map
+
+    if darkness is not None:
+        bg_data *= darkness
+    bg_color = plt.get_cmap('Greys')(bg_data)
+
+    # select vertices which are filtered out by the threshold
+    under_threshold = np.abs(surf_map) < (
+        absolute_threshold if absolute_threshold is not None else 0
+    )
+    # replace their color with the background color
+    vertexcolor[under_threshold] = bg_color[under_threshold]
+    # and merge background color with surface map color if need be
+    if bg_on_data:
+        vertexcolor[~under_threshold] = vertexcolor[~under_threshold] * \
+            bg_color[~under_threshold]
```

Comment:
multiplying the stat map color by the background color to mix them is a bit surprising  IMO. it only happens to work because here color channels are represented with floats in [0, 1] rather than eg 8-bit ints. also I wonder if a product is the right way to mix colors. For example if the background is black the stat map completely disappears (multiplied by 0), and it always results in darker colors -- eg if the red channel in both is .5 it will be 0.25 in the product. shouldn't we be using an average instead?

Reply:
Just for me to be sure: you suggest we do an average of the background color and the surface map's color, is that it? I think it's a good idea!

Reply:
Just for me to be sure: you suggest we do an average of the background color and the surface map's color, is that it? I think it's a good idea!

Reply:
I'm not sure an average is the right thing to do -- what we want to do is emulate transparency so we could look up how that  is usually done.

Reply:
> > I agree. Neither of these is great; "dim" sounds like "dimension"; "darkness" has a surprising behavior: increasing the darkness makes the image lighter; neither name hints that it is a scaling factor
> 
> Maybe `bg_brightness` / `bg_lighting` would be more intuitive?

`bg_brightness` is good I think.